h2o-py/tests/pyunit_utils/utils_for_glm_tests.py

```diff
@@ -0,0 +1,150 @@
+    if return_best:
+        print_model_hyperparameters(sortedGrid.models[0], hyper_parameters)
+        return sortedGrid.models[0]
+    else:
+        return grid_models_analysis(sortedGrid.models, hyper_parameters, metric=metric, epsilon=epsilon)
+
+def grid_models_analysis(grid_models, hyper_parameters, metric="logloss", epsilon=1e-3):
+    """
+    This method will search within the grid search models that have metrics within epsilon calculated as 
+    abs(metric1-metric2)/abs(metric1) as the best model.  We are wanting to send the best model that has the best
+     constraint values meaning either they have the smallest magnitude or if less than constraints, it has the smallest
+     magnitude and the correct sign.  Else, the original top model will be returned.
+    """
+    base_metric = grid_models[0].model_performance()._metric_json[metric]
+    base_constraints_table = grid_models[0]._model_json["output"]["linear_constraints_table"]
+    num_constraints = len(base_constraints_table.cell_values)
+    cond_index = base_constraints_table.col_header.index("condition")
+    [best_equality_constraints, best_lessthan_constraints] = grab_constraint_values(
+        base_constraints_table, cond_index, num_constraints)
+
+    base_iteration = find_glm_iterations(grid_models[0])
+    num_models = len(grid_models)
+    best_model_ind = 0
+    model_indices = []
+    model_equality_constraints_values = []
+    model_lessthan_constraints_values = []
+    iterations = []
+    for ind in range(1, num_models):
+        curr_model = grid_models[ind]
+        curr_metric = grid_models[ind].model_performance()._metric_json[metric]
+        metric_diff = abs(base_metric-curr_metric)/abs(base_metric)
+        if metric_diff < epsilon:
+            curr_constraint_table = curr_model._model_json["output"]["linear_constraints_table"]
+            [equality_constraints_values, lessthan_constraints_values] = grab_constraint_values(
+                curr_constraint_table, cond_index, num_constraints)
+            # conditions used to choose the best model
+            if (sum(equality_constraints_values) < sum(best_equality_constraints)) and (sum(lessthan_constraints_values) < sum(best_lessthan_constraints)):
+                best_model_ind = ind
+                base_iteration = find_glm_iterations(curr_model)
+                best_equality_constraints = equality_constraints_values
+                best_lessthan_constraints = lessthan_constraints_values
+            model_equality_constraints_values.append(equality_constraints_values)
+            model_lessthan_constraints_values.append(lessthan_constraints_values)
+            model_indices.append(ind)
+            iterations.append(find_glm_iterations(curr_model))
+    print("Maximum iterations: {0} and it is from model index: {1}".format(base_iteration, best_model_ind))
+    print_model_hyperparameters(grid_models[best_model_ind], hyper_parameters)
+    return grid_models[best_model_ind]
+
```

Comment:
```suggestion


```
Ideally, I would try to stick to PEP-8 and [keep 2 blank lines between top-level functions](https://peps.python.org/pep-0008/#blank-lines).